### PR TITLE
Challenge1/arakawa

### PR DESCRIPTION
**Pull request by rikky0611**

### CODE REVIEW

**Comment by JustGlowing** [comment]
Hi @rikky0611, I reviewed your submission and I have seen that you split the data in train set and test set then evaluated your regressor on a test set. This way you can't be sure that the accuracy you estimated on the test set is the accuracy that you should expect on new unseen data. Can you 

1) use cross validation on the training set to evaluate your regressor and compare the result with what you got on the test set? 
2) Explain why cross validation is more reliable than a simple train/test split?

More importantly, you should provide the MAPE on the test set provided by us.

**Comment by rikky0611** [comment]
@JustGlowing Thank you so much for the review. I read your review carefully and found some points I couldn't figure out. I split only train data ('data/training_dataset_500.csv') into train and valid data. I used valid data for validation of training(this time's MAPE value is just for check before prediction and different from the one in mape.txt). Then I used this trained model for prediction of test data('data/test_dataset_500.csv'), which I'd kept aside. I'm afraid that the MAPE value is from the provided test data.

But for the points you mentioned, I admit that I should have used cross validation because it gives me better accuracy regardless of how train/test_split split the data. It's effective when I have small data like this time.

I commit the notebook which includes cross validation.

**Comment by JustGlowing** [comment]
hi @rikky0611 thanks for your updates.

In the data folder, the file `test_dataset_500.csv` is provided. You have to evaluate your dataset on that data set.

Thanks for updating your code with the cross validation section. What can you conclude from the results that you got? Can you explain me why cross validation is better than a simple train test split?

**Comment by rikky0611** [comment]
@JustGlowing I'm sorry for a late reply. Thank you very much for commenting. From the cross-validation results, I could say my model has the accuracy around 13% for unseen data and it's not over-fitting. Cross validation is better than a simple train/test split in the following points.
* It gives me better accuracy for unseen data regardless of how train/test_split split the data. 
* I would get more data for training than normal split and an estimated error would be small even if I have small data. 

As for the test data set, I had loaded the 'data/test_dataset_500.csv' in the second cell and I predicted my model on that data, which I didn't touch at all.
'load'


'predict'

Would you point out my misunderstanding if any ?

**Comment by shirakia** [comment]
rikky0611 is now working with us!
